feat: support importing and ignoring non-evm tokens

juanmigdr opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
The feature to hide tokens is available for EVM assets but not for non-EVM assets. This task involves implementing a similar token hiding feature for non-EVM assets. This is crucial for improving user experience by allowing users to hide unwanted tokens, especially in light of spam and malicious token issues. The implementation should be prioritized to align with upcoming Solana campaigns.

Furthermore, having the possibility of hiding tokens is not great if you cannot add tokens back. Therefore I have implemented the whole logic to add non-VEVM tokens to your wallet and remove them from ignored tokens list.

In order to get this done the following PRs have been merged into Core:
- Support for adding non-evm tokens [here](https://github.com/MetaMask/core/pull/7016)
- Support for ignoring non-evm tokens [here](https://github.com/MetaMask/core/pull/6981)

Furthermore this PR solves a couple of bugs:
- Flashy behavior on the import token header when opening import token screen
- Already added tokens could be re-added which was incorrect, they have been disabled now

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: support importing and ignoring non-evm tokens

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/ASSETS-1425

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

https://github.com/user-attachments/assets/77de0410-53b3-43cc-a6e4-2951db6b2b32

### **After**

https://github.com/user-attachments/assets/629c34bd-e0a9-4eef-bf3c-ec03be874b1a

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Enables importing and hiding non‑EVM tokens, updates Add Asset/search UI to work across EVM/non‑EVM networks, and improves token removal, explorer links, and already‑added token handling.
> 
> - **Non‑EVM token support**:
>   - Add tokens via `MultichainAssetsController.addAssets` in `SearchTokenAutocomplete`; retain EVM via `TokensController.addTokens`.
>   - New `removeNonEvmToken` util to hide non‑EVM tokens; integrate in `Tokens` and `AssetOptions`.
>   - Disable re‑adding already‑added tokens (EVM via `selectTokensByChainIdAndAddress`, non‑EVM via `selectMultichainAssets`).
> - **Add Asset flow**:
>   - Use `useTopTokens` to source token lists; show loading state; only fetch token list for EVM.
>   - Render custom Token/NFT tabs only on EVM networks; support non‑EVM search/import.
>   - Network selector filters to EVM when not adding tokens.
> - **Search and lists**:
>   - Refactor `AssetSearch` to `useTokenSearch` with `allTokens` prop and debounced updates.
>   - `MultiAssetListItems` switches to `FlashList`, shows network badges, and disables already‑added tokens.
> - **Navigation/UI**:
>   - Update Import screens to use `getImportTokenNavbarOptions` (BottomSheetHeader).
>   - Enable "Add token" button on non‑EVM networks in `TokenListControlBar`.
> - **Asset details**:
>   - `AssetOptions` handles CAIP addresses for explorers, hides "Remove token" for native/wSOL, and supports non‑EVM removal with notifications/metrics.
>   - `ConfirmAddAsset` balances support non‑EVM addresses (hexified), and header updated.
> - **Tests**:
>   - Extensive unit/snapshot updates for new behaviors across search, add, list, confirm, tokens, and asset options.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f43d635e184c22c6902880f3380e7494843237ce. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

juanmigdr committed:
feat: support ignoring non-evm tokens

juanmigdr committed:
Merge branch 'main' into feat/support-ignore-non-evm

juanmigdr committed:
chore: updated to use nonEvmAccount instead of solana and hid ignore feature until we support token import

juanmigdr committed:
Merge branch 'main' into feat/support-ignore-non-evm

juanmigdr committed:
fix: token import flow is broken

juanmigdr committed:
chore: minor rollback

juanmigdr committed:
Merge branch 'fix/broken-import-token-flow' into feat/support-importing-non-evm-tokens

juanmigdr committed:
fix: select box is not clickable

juanmigdr committed:
chore: added support for searching and adding non-evm tokens and improved performance and bugs

juanmigdr committed:
chore: enabled button to add tokens on non-evm networks

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

juanmigdr committed:
Merge branch 'main' into feat/support-ignore-non-evm

juanmigdr committed:
chore: fix asset options issue

juanmigdr committed:
Merge branch 'feat/support-ignore-non-evm' into feat/support-importing-non-evm-tokens

juanmigdr committed:
chore: remove unecessary add asset

juanmigdr commented on app/component-library/components/List/ListItemMultiSelect/ListItemMultiSelect.types.ts:
```diff
@@ -18,6 +18,8 @@ export interface ListItemMultiSelectProps
    */
   isDisabled?: boolean;
+
+  onPress?: () => void;
```
Fix incorrect typings for ListMultiSelect

juanmigdr commented on app/components/UI/Tokens/TokenListControlBar/TokenListControlBar.tsx:
```diff
@@ -29,9 +29,7 @@ export const TokenListControlBar = ({
       size={ButtonIconSizes.Lg}
       onPress={goToAddToken}
       iconName={IconName.Add}
-      isDisabled={!isEvmSelected}
```
Allow adding non-evm tokens through the UI

juanmigdr committed:
chore: enable hiding tokens

juanmigdr committed:
chore: show action sheet to hide token for non-evm

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

juanmigdr committed:
chore: support adding multiple assets at once

juanmigdr committed:
chore: updated tests

juanmigdr committed:
chore: update tokens tests

juanmigdr committed:
fix: broken tests

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

juanmigdr committed:
fix: linting errors

juanmigdr committed:
chore: updated snapshot

juanmigdr committed:
chore: updated snapshot

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

juanmigdr committed:
chore: added tests

juanmigdr committed:
chore: more coverage

juanmigdr committed:
chore: increased test coverage

juanmigdr committed:
chore: improved coverage

juanmigdr committed:
chore: increase coverage

juanmigdr committed:
chore: remove unecessary tests

juanmigdr committed:
chore: minor changes

juanmigdr committed:
chore: minor change

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

juanmigdr committed:
fix: non-evm shown when adding nft

juanmigdr committed:
chore: simplify getImportTokenNavbarOptions

juanmigdr committed:
fix: flashy header in import flow

juanmigdr committed:
feat: disable adding already added tokens

juanmigdr committed:
chore: mino changes

juanmigdr committed:
chore udated some snapshots

juanmigdr committed:
chore: updated tests

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

Prithpal-Sooriya commented on app/components/UI/Tokens/index.test.tsx:
```diff
@@ -992,4 +1046,229 @@ describe('Tokens', () => {
+
+      // Verify Logger is available for error logging
+      expect(Logger.log).toBeDefined();
+      expect(typeof Logger.log).toBe('function');
```
Same, can we instead mock the logger and assert the logger was called?

Prithpal-Sooriya commented on app/components/UI/Tokens/index.test.tsx:
```diff
@@ -992,4 +1046,229 @@ describe('Tokens', () => {
+      });
+    });
+
+    it('navigates to AddAsset for non-EVM chains', async () => {
```
idk, should we have 2 tests (1 evm and 1 non-evm)? Both feel very similar, could be a single test?

Prithpal-Sooriya commented on app/components/UI/Tokens/index.test.tsx:
```diff
@@ -992,4 +1046,229 @@ describe('Tokens', () => {
+      // Verify component renders correctly with tokens
+      expect(
+        getByTestId(WalletViewSelectorsIDs.TOKENS_CONTAINER),
+      ).toBeOnTheScreen();
```
idk, do we need both these identical assertions?

Prithpal-Sooriya commented on app/components/Views/AddAsset/AddAsset.test.tsx:
```diff
@@ -403,4 +442,139 @@ describe('AddAsset component', () => {
+    it('displays loading indicator when useTopTokens pending is true', () => {
+      mockUseParamsValues.assetType = 'token';
+
+      (useTopTokens as jest.Mock).mockReturnValue({
```
nit - we can use this to create a mocked instance and reduce type assertions:
```ts
// This is also strictly typed (so mockReturnValue will provide type hints for the correct return type)
const mockUseTopTokens = jest.mocked(useTopTokens)
```

Prithpal-Sooriya replied:
meh, this ain't a blocker

Prithpal-Sooriya commented on app/components/Views/AssetOptions/AssetOptions.test.tsx:
```diff
@@ -402,4 +469,310 @@ describe('AssetOptions Component', () => {
+        />,
+      );
+
+      expect(getByText('Remove token')).toBeOnTheScreen();
```
idk if this assertion is good for this test. We're not really testing the correct display options for non-evm are we?

smilingkylan committed:
chore: Suppress CI error (#22280)

The purpose of this task is to fix the `yarn audi:ci` issues that popped
up for `react-native-community/cli` packages due to a recent
vulnerability related to bash commands being executed from within that
package

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds `npmAuditIgnoreAdvisories: [1109627]` to `.yarnrc.yml` to
temporarily suppress `@react-native-community/cli` audit failures and
unblock CI.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
65be2654ee7077fe52e03571639067e3186deaa9. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

---------

Co-authored-by: Luis Taniça <matallui@gmail.com>

caieu committed:
chore(predict): change polymarket links to use webview (#22255)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

- change polymarket links to use webview

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Routes Polymarket Terms and Resolution links to the in-app Webview
using navigation (with InteractionManager), replacing direct Linking
usage, and adds tests.
> 
> - **Predict UI**:
> - `PredictConsentSheet.tsx`: Replace `Linking.openURL` with in-app
Webview navigation via `useNavigation` and
`InteractionManager.runAfterInteractions`; wrap description/`Learn more`
in a `TouchableOpacity`.
> - `PredictMarketDetails.tsx`: Add handler to navigate to Webview for
Polymarket resolution details via
`InteractionManager.runAfterInteractions`.
> - **Tests**:
> - `PredictConsentSheet.test.tsx`: Mock
`InteractionManager.runAfterInteractions`; assert navigation to
`Webview/SimpleWebview` with Polymarket TOS URL and title.
> - `PredictMarketDetails.test.tsx`: Mock
`InteractionManager.runAfterInteractions`; verify About tab "Polymarket"
press navigates to `Webview/SimpleWebview` with resolution docs URL and
title.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
9f4e219566a70aff91ef186092eab10b35950754. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

kevinbluer committed:
fix(predict): Various chart UX fixes / improvements (#22220)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

- Various market details chart fixes / improvements 

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes: NA

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**


https://github.com/user-attachments/assets/665e3b10-522e-4a1b-ad0b-7d6253a99f43

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Refines PredictDetailsChart visuals and behavior (loading indicator,
grid/labels/insets), updates date/percent formatting, and adds
comprehensive utils tests.
> 
> - **UI/Chart UX**:
> - Replace loading placeholder text with `ActivityIndicator`; keep
chart skeleton visible during loading.
> - Tweak layout/spacing: adjust margins (`mb-6`→`mb-4/mb-3`), ensure
axis label row min-height, and legend spacing (`mb-3`→`mb-2`).
> - Update axis label style to `text-[11px]` and use alternative text
color.
> - Refine grid: shift lines (`x1` to `8`), move tick labels up (`y -
4`) and set color to `colors.text.alternative`.
>   - Adjust area gradient fade (`Stop offset` from `100%` to `95%`).
> - **Chart behavior/constants** (`utils.ts`):
> - Change `LINE_CURVE` alpha to `0.2`; update `CHART_CONTENT_INSET` to
`{ top: 8, bottom: 4, left: 8, right: 48 }`.
> - Revise label formatting: `ONE_WEEK` now `weekday AM/PM`; `MAX` uses
2‑digit year; support seconds/milliseconds.
> - Add `formatTickValue` precision rules based on range (<1: 2dp, <10:
1dp, ≥10: 0dp).
> - **Tests**:
>   - Update chart tests to check for `line-chart` during loading.
> - Add comprehensive `utils.test.ts` covering constants, date label
formatting, and tick value formatting edge cases.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
2f1e80afddb689e230ff4013081456c04a262ada. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

---------

Co-authored-by: Luis Taniça <matallui@gmail.com>

matallui committed:
feat(predict): relay sell orders (#22290)

## **Description**

This PR enables relaying of SELL orders through the MetaMask Predict API
relay, similar to how BUY orders are already relayed. Previously, only
BUY orders were sent through the relay to collect fees, while SELL
orders went directly to the Polymarket CLOB endpoint.

**Reason for change:**
- SELL orders should also be relayed to enable consistent fee collection
and monitoring across all order types
- Provides better control and observability for all trading activity

**Improvements:**
- All orders (BUY and SELL) now route through the `CLOB_RELAYER`
endpoint
- Rate limiting now applies to both BUY and SELL orders based on the
last BUY order timestamp
- Simplified order submission logic by removing conditional endpoint
selection
- Updated `OrderResponse` type fields to be optional for better error
handling
- Renamed `BUY_ORDER_RATE_LIMIT_MS` to `ORDER_RATE_LIMIT_MS` to reflect
that rate limiting applies to all orders

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: N/A

## **Manual testing steps**

```gherkin
Feature: Polymarket SELL order relaying

  Scenario: user places a SELL order
    Given user has an open position in a Polymarket market
    And user is connected to Polygon mainnet
    And the Predict feature is enabled

    When user navigates to the position details
    And user initiates a SELL order
    And user confirms the transaction
    Then the order should be relayed through the MetaMask API endpoint
    And the order should be successfully submitted to Polymarket
    And fees should be collected appropriately

  Scenario: user is rate limited after placing orders
    Given user has placed a BUY order within the last 5 seconds

    When user attempts to preview a new order (BUY or SELL)
    Then the preview should show rateLimited: true
    And user should be prevented from placing another order too quickly
```

## **Screenshots/Recordings**

N/A - Backend/API changes only, no UI changes

### **Before**

- BUY orders → CLOB_RELAYER endpoint (with fees)
- SELL orders → Direct to Polymarket CLOB endpoint (no fees)
- Rate limiting only applied to BUY order previews

### **After**

- BUY orders → CLOB_RELAYER endpoint (with fees)
- SELL orders → CLOB_RELAYER endpoint (no fees)
- Rate limiting applies to all order previews after a BUY order

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I've applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Routes both BUY and SELL orders through the MetaMask Predict relayer,
applies unified rate limiting, and updates response handling/types with
corresponding tests.
> 
> - **Order routing (utils/submitClobOrder)**:
> - Always post to `CLOB_RELAYER` (`/order`), removing conditional CLOB
endpoint usage.
> - Send both underscore and dash variants of `POLY_*` headers; include
`feeAuthorization` when provided.
> - **Rate limiting**:
> - Rename `BUY_ORDER_RATE_LIMIT_MS` to `ORDER_RATE_LIMIT_MS` and apply
in `PolymarketProvider.isRateLimited`.
> - `previewOrder` now rate-limits when a `signer` is present (affects
SELL after a recent BUY).
> - **Order handling**:
> - `OrderResponse` fields (`errorMsg`, `makingAmount`, `orderID`,
`status`, `takingAmount`, `transactionsHashes`) made optional.
> - `placeOrder` checks top-level `success` and nested
`response.success`; returns `errorMsg` when present.
> - **Tests**:
> - Update submit order tests to expect relayer URL, dual-format
headers, and serialized `feeAuthorization`.
> - Adjust expectations for `response` shape (includes `success: true`).
> - Update rate limit test to expect SELL previews to be rate-limited
after a BUY.
>   - Remove unused `errorCode` from 403 path expectations.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
33a2d3e3937563d55a98293e137570b8f1852f87. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

caieu committed:
chore(predict): cash out screen changes (#22272)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

- cash out changes

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates the Predict cash out preview to show per‑share price with
fallback, localizes copy, refactors layout/styles, adds locale strings,
and adjusts tests accordingly.
> 
> - **Predict Sell Preview (UI/Logic)**:
> - Localizes header/button text and replaces hardcoded labels with
`strings('predict.cash_out')`.
> - Displays per‑share price using `preview.sharePrice ??
position.price` and adds line under current value via
`predict.at_price_per_share`.
> - Shows initial position info (amount, outcome, avg entry price) via
`predict.cashout_info`.
> - Refactors layout to `Box` components, tweaks icon container, and
simplifies details rendering.
> - **State/Computation**:
> - Adds `currentPrice` derived from `preview?.sharePrice ??
position?.price`; leverages `position.avgPrice` for display.
> - **Styles**:
> -
`app/components/UI/Predict/components/PredictPosition/PredictPosition.styles.ts`:
make `positionDetails` `flex: 1`; remove extra width/flex on
`positionPnl`.
> -
`app/components/UI/Predict/views/PredictSellPreview/PredictSellPreview.styles.ts`:
remove unused containers/fields; keep core layout and button styles.
> - **Tests** (`PredictSellPreview.test.tsx`):
> - Update assertions to new copy (“Cash out”, `$50.00 on Yes at 50¢`),
and add test for undefined `preview.sharePrice` using fallback.
> - **Locales** (`locales/languages/en.json`):
> - Add `predict.at_price_per_share` and `predict.cashout_info`; ensure
“Cash out” casing alignment.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
fae7da89abfeb71b357389b03ab7342744ec5f30. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

---------

Co-authored-by: Luis Taniça <matallui@gmail.com>

kevinbluer committed:
fix(predict): Resolves search issue (removes spurious querystring param) (#22271)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

- Resolves an issue with the search wherein inactive markets were coming
back from the `/public-search` endpoint due to a spurious querystring
param being passed

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:
[PRED-264](https://consensyssoftware.atlassian.net/browse/PRED-264)

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<img width="420" alt="image"
src="https://github.com/user-attachments/assets/255c6be7-c3d0-4d05-b913-627b16ee9d84"
/>


<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.


[PRED-264]:
https://consensyssoftware.atlassian.net/browse/PRED-264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Removes the extra `presets=Events` from the Polymarket `public-search`
request and updates tests to expect the new URL.
> 
> - **Polymarket search**:
> - Adjust `getParsedMarketsFromPolymarketApi` to build `public-search`
URL without `presets=Events`, keeping only `presets=EventsTitle`.
> - **Tests**:
> - Update expected URL in `utils.test.ts` to match the revised
`public-search` query (no `presets=Events`).
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
1d9ea65172429d2dc1cae8fd0fb3d4027e328f89. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Co-authored-by: Luis Taniça <matallui@gmail.com>

juanmigdr committed:
chore: solved some comments

juanmigdr committed:
chore improved readability

juanmigdr committed:
Merge branch 'main' into feat/support-importing-non-evm-tokens

Prithpal-Sooriya reviewed (approved):
Amazing work!
